.github/workflows/nightly.yml

```diff
@@ -109,3 +112,125 @@ jobs:
           name: darktable.${{ github.sha }}.win64.zip
           path: ${{ env.BUILD_DIR }}/darktable-*.exe
           retention-days: 2
+  
+  macOS:
+    name: Nightly darktable.${{ github.sha }}-${{ matrix.build.deployment }}-${{ matrix.build.xcode }}.macOS
+    runs-on: ${{ matrix.build.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        build:
+          - { os: macos-10.15, xcode: 11.3.1, deployment: 10.7 }
+          #- { os: macos-10.15, xcode: 11.7,   deployment: 10.14 } build fails
+          - { os: macos-10.15, xcode: 12.4,   deployment: 10.15 }
+          - { os: macos-11,    xcode: 12.5.1, deployment: 11.3 }
+        compiler:
+          - { compiler: XCode,   CC: cc, CXX: c++ }
+        btype: [ RelWithDebInfo ]
+        target:
+          - skiptest
+    env:
+      DEVELOPER_DIR: /Applications/Xcode_${{ matrix.build.xcode }}.app/Contents/Developer
+      CC: ${{ matrix.compiler.CC }}
+      CXX: ${{ matrix.compiler.CXX }}
+      MACOSX_DEPLOYMENT_TARGET: ${{ matrix.build.deployment }}
+      CXXFLAGS: -stdlib=libc++
+      OBJCXXFLAGS: -stdlib=libc++
+      SRC_DIR: ${{ github.workspace }}/src
+      BUILD_DIR: ${{ github.workspace }}/build
+      INSTALL_PREFIX: ${{ github.workspace }}/install
+      MACOS_PACKAGING_DIR: ${{ github.workspace }}/src/packaging/macosx
+      LENSFUN_INSTALL_PREFIX: ${{ github.workspace }}/lensfun
+      CMAKE_BUILD_TYPE: ${{ matrix.btype }}
+      #GENERATOR: Ninja
+      GENERATOR: Unix Makefiles
+      TARGET: ${{ matrix.target }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+          path: src
+      - name: Install Base Dependencies
+        run: |
+          brew update > /dev/null || true
+          brew unlink python@3.8
+          brew unlink gcc@9
+          sudo rm '/usr/local/bin/2to3'
+          brew link --overwrite python@3.9 # workaround introduced 30.12.2020, replace asap.
+          # brew upgrade --ignore-pinned # workaround introduced 18.07.2021, replace asap
+          brew tap Homebrew/bundle
+          cd src/.ci 
+          brew bundle --verbose
+          brew uninstall lensfun
```

Comment:
remove broken brew-based lensfun version. it might be worth considering to also build have any CI build use a self-compiled lensfun-version so that the builds resemble official builds as much as possible.
see steps https://github.com/darktable-org/darktable/pull/10360/files#diff-0d5658b415099a82c11c03a06ca4ec765b4003a1f4b2f3f1943980a882cf8aa6R167-R182)